tests/functional/u/undefined/undefined_variable_py38.py

```diff
@@ -155,3 +155,17 @@ def __init__(self, value):
 
 
 dummy = Dummy(value=val if (val := 'something') else 'anything')
+
+def expression_in_ternary_operator_inside_container():
+    """Named expression in ternary operator: inside container"""
+    return [val2 if (val2 := 'something') else 'anything']
+
+
+def expression_in_ternary_operator_inside_container_tuple():
+    """Same case, using a tuple inside a 1-element list"""
+    return [(val3, val3) if (val3 := 'something') else 'anything']
+
+
+def expression_in_ternary_operator_inside_container_wrong_position():
+    """2-element list where named expression comes too late"""
+    return [val3, val3 if (val3 := 'something') else 'anything']  # [used-before-assignment]
```

Comment:
Pff... Man that's convoluted. Can't we raise an `don't-write-code-like-this` warning here? 😅

Reply:
Haha, I had to take 2-3mn and run the code to understand myself 😄 But do we really want to make the kind of person that would write this, flock to our bug tracker to complain 😉 ?